Add active Field To Project Versions

aravindparappil46 opened the pull request:
### Description
In order for the frontend to distinguish between active or inactive project versions, we need to include the `active` boolean field to `ProjectVersion` model.

<!-- REQUIRED
    Provide a concise description of your change. What does it do? Why is it necessary?
    As a guideline, think about how you would describe your change if you were to write a changelog entry for it.
-->

### Addressed Issue
Required for https://github.com/DependencyTrack/frontend/issues/844

<!-- REQUIRED
    Reference the issue addressed by this PR, e.g. `#1234`.
    Use keywords like `closes` or `fixes` to signal that this PR resolves the issue,
    causing the issue to be closed automatically when the PR is merged:
        https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword
-->

<!-- OPTIONAL
    If desired, share more technical details about the change here.
    Elaborating on why you implemented the change the way you did can be super helpful to the reviewer.
    Did you consider other solutions? Any problems you ran into along the way?
-->

### Checklist

<!-- REQUIRED
    Mark items in this list as done by adding a `x` between the square brackets.
    Non-applicable items may be marked as such by surrounding their text with tildes (`~`).

    This is not meant to be a strict to-do list. If you're unsure about anything,
    just leave it empty for now. The maintainers are happy to assist you in figuring it out!
-->

- [x] I have read and understand the [contributing guidelines](../CONTRIBUTING.md#pull-requests)
- [ ] This PR fixes a defect, and I have provided tests to verify that the fix is effective
- [x] This PR implements an enhancement, and I have provided tests to verify that it works as intended
- [ ] This PR introduces changes to the database model, and I have added corresponding [update logic](https://github.com/DependencyTrack/dependency-track/tree/master/src/main/java/org/dependencytrack/upgrade)
- [ ] This PR introduces new or alters existing behavior, and I have updated the [documentation](https://github.com/DependencyTrack/dependency-track/tree/master/docs/_docs) accordingly

nscuro commented:
@aravindparappil46 The coverage check is failing, primarily because the change itself is so tiny. 

We can bypass it, but another solution could be to remove the `set*` methods from the `ProjectVersion` class. DataNucleus can use the constructor to instantiate the class. It seems the missing coverage here comes from the fact that the `setActive` method is never called.

aravindparappil46 commented:
Thank you @nscuro ! I removed the `setActive` which I added and the coverage check is passing now.
Do let me know if you'd prefer removing all of the setters _(I didn't want to refactor parts I didn't touch without confirming)_

nscuro commented:
> Do let me know if you'd prefer removing all of the setters [...]

I think that would be good to avoid future confusion. You could also simply convert `ProjectVersion` to a [`record`](https://docs.oracle.com/en/java/javase/21/language/records.html) instead.
